fix_includes.py

```diff
@@ -1339,87 +1339,160 @@ def _GetFirstNamespaceLevelReorderSpan(file_lines):
+      elif line_info.type == _ENDIF_RE:
+        ifdef_depth -= 1
+
+      elif ifdef_depth != 0:
+        continue # skip lines until we're outside of an ifdef block
+
+      # Build the simplified namespace lists and namespace depths.  When
+      # any new namespace is encountered, add the namespace to the list
+      # using the next line to cover namespaces without forward declares.
+      # When a forward declare is found, add the concatenated namespaces
+      # to the list.  Once a contentful line (None) has been found, add
+      # the final namespace if we're still inside one and stop searching.
+      elif line_info.type == _NAMESPACE_START_RE:
+        for namespace in _GetNamespaceNames(line_info.line):
+          if not namespace:
+            namespace_prefixes.append('namespace {')
+          else:
+            namespace_prefixes.append('namespace %s {' % namespace)
+        
+        namespace_reorder_spans[' '.join(namespace_prefixes)] = (
+          line_number+1, line_number+1)
 
-    else:
-      # We should have handled all the cases above!
-      assert False, ('unknown line-info type',
-                     _LINE_TYPES.index(line_info.type))
+      elif line_info.type == _NAMESPACE_START_ALLMAN_RE:
+        pending_namespace_prefix = ''
+        namespaces = _GetNamespaceNames(line_info.line)
+        if len(namespaces) != 1:
+          raise FixIncludesError('Allman namespace found containing multiple '
+                                 'names: %s', line_info.line)
+        for namespace in namespaces:
+          if not namespace:
+            pending_namespace_prefix += 'namespace'
+          else:
+            pending_namespace_prefix += 'namespace %s' % namespace
+
+      elif line_info.type == _NAMESPACE_START_MIXED_RE:
+        # For mixed namespace styles, we need to append normalized prefixes
+        # using regular and Allman style.  Treat the first elements as
+        # normal and only treat the final element as Allman.  By the
+        # nature of mixed namespaces, there will always be more than
+        # one namespace so it is okay to assume that _GetNamespaceNames
+        # will always return multiple records.
+        pending_namespace_prefix = ''
+        namespaces = _GetNamespaceNames(line_info.line)
+        for namespace in namespaces[:-1]:
+          if not namespace:
+            namespace_prefixes.append('namespace {')
+          else:
+            namespace_prefixes.append('namespace %s {' % namespace)
+
+        if not namespaces[-1]:
+          pending_namespace_prefix += 'namespace'
+        else:
+          pending_namespace_prefix += 'namespace %s' % namespaces[-1]
+
+      elif line_info.type == _NAMESPACE_CONTINUE_ALLMAN_MIXED_RE:
+        # Append to the simplified allman namespace.
+        if pending_namespace_prefix == '':
+          raise FixIncludesError('Namespace bracket found without an associated '
+                                 'namespace name at line: %s', line_number)
```

Comment:
`line_info.line` or `line_number + 1`?

Reply:
I was going to make it line_info.lin but I started thinking that it would only print out `{` in most cases which might not be very helpful.  I opted for line number so you'd be able to see what line number the bracket in question would be.  I don't think I'd need a +1 since line 0 is marked as deleted on the script and just gets skipped over immediately so line_number actually corresponds to the actual line number (context shift for a computer scientist haha).

Reply:
Oh, I thought `line_info.line` *was* the line number, not the line content.

As for `line_number`, it's consistently used with +1 through the function, should that also be removed, then? (I suggested `enumerate(lines, start=1)` to get rid of this earlier.

Reply:
@kimgr the line_number+1 is actually meant to include the next line in the output as the possible reorder span.  For example
```
namespace A // Begin Allman if statement
{ // Begin continue bracket and add the next line as the possible reorder span, not this line
class Forwared_declared_class; // line_number+1 from line above as the possible reorder span
```
If I did enumerate with start=1, I'd be shifting the actual line numbers than from what the script initially did when it registered each line to a regex.  I'll add comments to the sections where I'm adding line_number+1.

Reply:
Oh, I see! I had misunderstood the `line_number + 1` as an adjustment from zero-based to one-based. Case closed, thanks!